mllib/src/main/scala/org/apache/spark/ml/clustering/GaussianMixture.scala

```diff
@@ -486,6 +491,9 @@ class GaussianMixture @Since("2.0.0") (
 @Since("2.0.0")
 object GaussianMixture extends DefaultParamsReadable[GaussianMixture] {
 
+  /** Limit number of features such that numFeatures^2^ < Integer.MaxValue */
+  private[clustering] val MAX_NUM_FEATURES = 46000
```

Comment:
In https://github.com/apache/spark/pull/15413, the symmetry of covariance matrix is taken into account and only the upper triangular part is store. So this number seems to be 65535? (`math.sqrt(Int.MaxValue.toDouble * 2)`)